aws/data_source_aws_organizations_organizational_unit_test.go

```diff
@@ -0,0 +1,56 @@
+package aws
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceAwsOrganizationsOrganizationalUnit_basic(t *testing.T) {
+	resourceName := "aws_organizations_organizational_unit.test_ou2"
+	dataSourceName := "data.aws_organizations_organizational_unit.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsOrganizationsOrganizationalUnitConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(resourceName, "name", dataSourceName, "children.0.name"),
+					resource.TestCheckResourceAttrPair(resourceName, "id", dataSourceName, "children.0.id"),
+					resource.TestCheckResourceAttrPair(resourceName, "arn", dataSourceName, "children.0.arn"),
+				),
+				/* We ExpectNonEmptyPlan due to the explicit datasource dependency on the test_ou2 resource.
+				 * See Terraform config comments for more details. */
+				ExpectNonEmptyPlan: true,
```

Comment:
This can be removed with the change to the implicit attribute reference to `aws_organizations_organizational_unit.test_ou2.parent_id`